File: backend/test_ws.js
const ws = new WebSocket('ws://127.0.0.1:8000/ws/live?agent_id=kizuna&lang=en');

ws.addEventListener('open', () => {
  console.log('connected');
});

ws.addEventListener('close', (event) => {
  console.log('disconnected', event.code, event.reason);
});

ws.addEventListener('message', (event) => {
  console.log('message', event.data);
});

ws.addEventListener('error', (event) => {
  console.log('error', event.error);
});

File: backend/test_ws.py
import asyncio
import websockets

async def test_ws():
    uri = 'ws://127.0.0.1:8000/ws/live?agent_id=kizuna&lang=en'
    print(f'Connecting to {uri}...')
    try:
        async with websockets.connect(uri) as websocket:
            print('Connected!')
            while True:
                msg = await websocket.recv()
                print(f'Received: {msg}')
    except Exception as e:
        print(f'Connection failed: {e}')

asyncio.run(test_ws())
